test/unit/account/states_test.rb

```diff
@@ -213,7 +215,9 @@ def test_schedule_for_deletion
 
       Timecop.freeze do
         account.schedule_for_deletion!
-        assert_equal Time.zone.now.beginning_of_day, account.reload.deleted_at
+        account.reload
+        assert_equal Time.zone.now.to_s, account.deleted_at.to_s
+        assert_equal Time.zone.now.to_s, account.state_changed_at.to_s
```

Comment:
Better save `Time.zone.now` to a variable, stub it before `schedule_for_deletion!` and assert the same value.